x/claims/keeper/integration_test.go

```diff
@@ -0,0 +1,525 @@
+			balanceClaims := s.app.BankKeeper.GetBalance(s.ctx, claimsAddr, claimsDenom)
+			Expect(balanceClaims.Amount.Uint64()).To(Equal(uint64(totalClaimsAmount - totalClaimed - remainderUnclaimed)))
+		})
+	})
+
+	Context("claiming after decay duration", func() {
+		BeforeAll(func() {
+			duration := params.AirdropEndTime().Sub(s.ctx.BlockHeader().Time)
+			s.CommitAfter(duration)
+			proposalId = govProposal(priv0)
+
+			// ensure module account has the unclaimed amount before airdrop ends
+			moduleBalance := s.app.ClaimsKeeper.GetModuleAccountBalances(s.ctx)
+			Expect(moduleBalance.AmountOf(claimsDenom).Uint64()).To(Equal(uint64(totalClaimsAmount - totalClaimed - remainderUnclaimed)))
+
+			// ensure community pool has 0 funds before airdrop ends
+			poolBalance := s.app.BankKeeper.GetBalance(s.ctx, distrAddr, claimsDenom)
+			Expect(poolBalance.Amount.Uint64()).To(Equal(uint64(remainderUnclaimed)))
+
+			s.Commit()
+		})
+
+		It("cannot claim after decay duration with partial claim", func() {
+			addr := getAddr(privs[2])
+			prebalance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			delegate(privs[2], 1)
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			Expect(balance.Amount.Uint64()).To(Equal(prebalance.Amount.Uint64() - 1))
+		})
+
+		It("dust amount removed from accounts with 0 claimed", func() {
+			addr := getAddr(privs[3])
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			Expect(balance.Amount.Uint64()).To(Equal(uint64(0)))
+
+		})
+
+		It("final balances", func() {
+			addr := getAddr(privs[0])
+			finalBalance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			claimed := 2*actionValue + actionValue/3
+			Expect(finalBalance.Amount.Uint64()).To(Equal(uint64(initClaimsAmount + claimed - 1)))
+
+			addr = getAddr(privs[1])
+			finalBalance = s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			claimed = actionValue + actionValue*2/3
+			Expect(finalBalance.Amount.Uint64()).To(Equal(uint64(initClaimsAmount + claimed - 2)))
+
+			addr = getAddr(privs[2])
+			finalBalance = s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			claimed = actionValue * 1 / 3
+			Expect(finalBalance.Amount.Uint64()).To(Equal(uint64(initClaimsAmount + claimed - 1)))
+
+			addr = getAddr(privs[3])
+			finalBalance = s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			Expect(finalBalance.Amount.Uint64()).To(Equal(uint64(0)))
+		})
+
+		It("can clawback unclaimed", func() {
+			// ensure module account is empty
+			moduleBalance := s.app.ClaimsKeeper.GetModuleAccountBalances(s.ctx)
+			Expect(moduleBalance.AmountOf(claimsDenom).Uint64()).To(Equal(uint64(0)))
+
+			// The unclaimed amount goes to the community pool
+			poolBalance := s.app.BankKeeper.GetBalance(s.ctx, distrAddr, claimsDenom)
+			expectedBalance := totalClaimsAmount - totalClaimed
+			Expect(poolBalance.Amount.Uint64()).To(Equal(uint64(expectedBalance)))
```

Comment:
This fails with
```
Expected
      <uint64>: 3116668
  to equal
      <uint64>: 2916668
  In [It] at: /home/runner/work/evmos/evmos/x/claims/keeper/integration_test.go:334
```
`2916668` is the correct amount that the community pool needs to receive from the unclaimed amounts.
So, I assume I need to apply the formula for inflation.

Reply:
I tried it in the last commit: https://github.com/tharsis/evmos/pull/373/files#diff-4bffcbe1b766a74f99b02c3e9ea3e9e97bfa79c280ffaadd99e679ea5a838bbbR92, but it does not solve the final balance calculation. So, the problem might be elsewhere.

Reply:
Solved. The difference was the "dust" amount given for performing the claim, that is also clawed back if not used.